src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -1061,12 +1073,14 @@ void Esp32WiFiManager::start_ssid_scan(Notifiable *n)
     ESP_ERROR_CHECK(esp_wifi_scan_start(&cfg, false));
 }
 
+// Returns the number of SSIDs found in the last scan.
 size_t Esp32WiFiManager::get_ssid_scan_result_count()
 {
     OSMutexLock l(&ssidScanResultsLock_);
     return ssidScanResults_.size();
 }
 
+// Returns one SSID record from the last scan.
 const wifi_ap_record_t& Esp32WiFiManager::get_ssid_scan_result(size_t index)
 {
     OSMutexLock l(&ssidScanResultsLock_);
```

Comment:
fyi this code is not correct, because the mutex is released but a reference to the storage area is returned to the caller.

Reply:
Ok, let me send in an update PR with this changed to return by value.